okapi-common/src/main/java/org/folio/okapi/common/ModuleId.java

```diff
@@ -99,12 +143,24 @@ public String toString() {
     return b.toString();
   }
 
+  /**
+   * static comparsion
+   * @param i1 left
+   * @param i2 right
+   * @return See
+   * {@link org.folio.okapi.common.ModuleId#compareTo(org.folio.okapi.common.ModuleId)
+   */
   public static int compare(String i1, String i2) {
     ModuleId m1 = new ModuleId(i1);
     ModuleId m2 = new ModuleId(i2);
     return m1.compareTo(m2);
   }
 
+  /**
+   * Returns newest module out of a list of modules
+   * @param l list of module IDs
+   * @return newest module (null for empty list)
```

Comment:
No, it never returns null.
It first filters all modules where the procuct equals to this.getProduct(). If there is none it returns this.getId() otherwise it returns the newest of them.

Reply:
damn right

Reply:
I sort of regret this feature.. It's offerd by Okapi so you can supply a fitler.. Eg "foo" will match module ID with product "foo". And "foo-1" will match anything with product "foo" and major=1.

Reply:
There's no such notion in semver